eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -321,18 +323,29 @@ stages:
               -PRTitle "$(PrTitle)-generated-from-$(Build.DefinitionName)-$(Build.BuildId)"
               -PRBody "$(GeneratedSDKInformation)  $(ReleasePlanInfo)"
               -OpenAsDraft $true
-
-        - pwsh: |
-            Write-Host "Release Info: $(ReleasePlanInfo)"
-            $prStatus = 'Failed to generate SDK.'
-            Write-Host "##vso[task.setvariable variable=SdkPrStatus]$prStatus"
-          displayName: 'Set PR generation status'
-          condition: Failed()
+        
 
         - ${{ if ne(parameters.ReleasePlanWorkItemId, 0) }}:
+          - pwsh: |
+              $prUrl = "https://github.com/Azure/$(SdkRepoName)/pull/$(Submitted.PullRequest.Number)"
+              Write-Host "Pull request created: $prUrl"
+              Write-Host "##vso[task.setvariable variable=PullRequestUrl]$prUrl"
+              $prStatus = "draft"
+              Write-Host "##vso[task.setvariable variable=SdkPrStatus]$prStatus"
+            condition: and(succeeded(), eq(variables['HasChanges'], 'true'), ne(variables['Build.Reason'], 'PullRequest'), not(endsWith(variables['SdkRepoName'], '-pr')))
+            displayName: "Set pull request URL variable"
+
+          - pwsh: |
+              $prUrl = ""
+              Write-Host "##vso[task.setvariable variable=PullRequestUrl]$prUrl"
+              $prStatus = "Failed to generate SDK."
+              Write-Host "##vso[task.setvariable variable=SdkPrStatus]$prStatus"
+            condition: failed()
+            displayName: "Set pull request generation failed status variable"
+
           - task: AzureCLI@2
             displayName: Link pull request to release plan
-            condition: and(succeededOrFailed(), eq(variables['HasChanges'], 'true'), not(endsWith(variables['SdkRepoName'], '-pr')))
+            condition: not(endsWith(variables['SdkRepoName'], '-pr'))
```

Comment:
Why did this condition change? It should still ony trigger if it haschanges right?

Reply:
I guess you always want to set the status even if there is no PR?